src/components/TodoList/TodoList.tsx

```diff
@@ -1,12 +1,41 @@
 /* eslint-disable */
 import React from 'react';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { Todo } from '../../types/Todo';
+import { actions as currTodosActions } from '../../features/currentTodo';
 
 export const TodoList: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const todos = useAppSelector(state => state.todos);
+  const currentFilter = useAppSelector(state => state.filter);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+
+  const handleChooseTodo = (todo: Todo) => {
+    dispatch(currTodosActions.setTodo(todo));
+  };
+
+  const visibleTodos = todos.filter(todo => {
+    const matchesQuery = todo.title
+      .toLowerCase()
+      .includes(currentFilter.query.toLowerCase());
+
+    switch (currentFilter.status) {
+      case 'completed':
+        return todo.completed && matchesQuery;
+      case 'active':
+        return !todo.completed && matchesQuery;
+      default:
+        return matchesQuery;
+    }
+  });
```

Comment:
No need to filter in case Filter is all or query is empty - it may cause performance issues